packages/plugins/io-center/src/ui/components/canvas/canvas-area.svelte

```diff
@@ -21,9 +22,10 @@
 		onAddLC: (type: LcTypes, number?: number) => void;
 		editLC: (lcNode: NodeElementType, newType: LcTypes) => void;
 		hasLNodeType: (type: LcTypes) => boolean;
+		addConnection: (connection: Connection) => void;
 	};
 
-	const { onAddLC, editLC, hasLNodeType }: Props = $props();
+	const { onAddLC, editLC, hasLNodeType, addConnection }: Props = $props();
 
 	let isDialogOpen = $state(false);
 </script>
```

Comment:
I was not inspired for the naming, I let you decide...

```suggestion
async function currentCoordinates(connection: Connection) {
	const fromXToY = `${(await getCoordinates(connection.from)).x},${(await getCoordinates(connection.from)).y}`
	const fromXToXFromY = `${((await getCoordinates(connection.from)).x + (await getCoordinates(connection.to)).x) / 2},${(await getCoordinates(connection.from)).y}`
	const fromXToXToY = `${((await getCoordinates(connection.from)).x + (await getCoordinates(connection.to)).x) / 2},${(await getCoordinates(connection.to)).y}`
	const toXToY = `${(await getCoordinates(connection.to)).x},${(await getCoordinates(connection.to)).y}`
	return `M ${fromXToY} C ${fromXToXFromY} ${fromXToXToY} ${toXToY}`
}
</script>
```